### ISSUE TITLE
Store semantic values in all AST nodes

### ISSUE THREAD

**Issue by aslakhellesoy** [cucumber-attic/bool#39]
Example: When we create a new `And` node, we should not only store the `left` and `right` nodes, but also the [semantic value](http://www.gnu.org/software/bison/manual/html_node/Semantic-Values.html) of the token (`&&`).

This is not strictly needed for bool (although it could be used in the renderer), but it is going to be necessary for gherkin3 when we render features in reports.

**Comment by aslakhellesoy** [comment]
Yes, that's it. We need to store it since we have so many translated variants.

**Comment by aslakhellesoy** [comment]
What I pushed to the `semantic-values` branch is probably garbage. If you want to work on this, you could start by modifying the AST nodes to include the value. For example (`ast.js`):

``` javascript
module.exports = {
  Var: function Var(value) {
    this.value = value;

    this.accept = function(visitor, args) {
      return visitor.visit_var(this, args);
    }
  },

  And: function And(value, left, right) {
    this.value = value;
    this.left = left;
    this.right = right;

    this.accept = function(visitor, args) {
      return visitor.visit_and(this, args);
    }
  },

  Or: function Or(value, left, right) {
    this.value = value;
    this.left = left;
    this.right = right;

    this.accept = function(visitor, args) {
      return visitor.visit_or(this, args);
    }
  },

  Not: function Not(value, operand) {
    this.value = value;
    this.operand = operand;

    this.accept = function(visitor, args) {
      return visitor.visit_not(this, args);
    }
  }
};
```

And `renderer.js`:

``` javascript
module.exports = function Renderer() {
  this.visit_var = function(node, vars) {
    return node.value;
  };

  this.visit_and = function(node, vars) {
    return "(" + node.left.accept(this, vars) + " " + node.value + " " + node.right.accept(this, vars) + ")";
  };

  this.visit_or = function(node, vars) {
    return "(" + node.left.accept(this, vars) + " " + node.value + " " + node.right.accept(this, vars) + ")";
  };

  this.visit_not = function(node, vars) {
    return node.value + node.operand.accept(this, vars);
  };
};
```

**Comment by aslakhellesoy** [comment]
I have a fix for this in #47. Please review.

**Comment by aslakhellesoy** [comment]
Fixed on master
